src/core/default_appearance.js

```diff
@@ -87,6 +87,94 @@ function parseDefaultAppearance(str) {
   return new DefaultAppearanceEvaluator(str).parse();
 }
 
+class AppearanceEvaluator extends EvaluatorPreprocessor {
+  constructor(str) {
+    super(str);
+    this.str = str;
+  }
+
+  parse() {
+    const operation = {
+      fn: 0,
+      args: [],
+    };
+    const result = {
+      fontSize: 0,
+      fontName: "",
+      fontColor: /* black = */ new Uint8ClampedArray(3),
+    };
+
+    let isInTextSection = false;
+    let scaleFactor = 1;
+
+    try {
+      while (true) {
+        operation.args.length = 0; // Ensure that `args` it's always reset.
+
+        if (!this.read(operation)) {
+          break;
+        }
+        const { fn, args } = operation;
+
+        switch (fn | 0) {
+          case OPS.beginText:
+            isInTextSection = true;
+            break;
+          case OPS.endText:
+            isInTextSection = false;
+            scaleFactor = 1;
+            break;
+          case OPS.setTextMatrix:
+            scaleFactor *= Math.abs(args[0]);
```

Comment:
Also, isn't it possible that other components of the textMatrix also affects the size?